src/playlist.js

```diff
@@ -259,33 +136,14 @@
       return videojs.createTimeRange(0, duration(playlist));
     }
 
-    start = 0;
-    end = intervalDuration(playlist,
-                           playlist.mediaSequence,
-                           playlist.mediaSequence + playlist.segments.length);
-    targetDuration = playlist.targetDuration || DEFAULT_TARGET_DURATION;
-
     // live playlists should not expose three segment durations worth
     // of content from the end of the playlist
     // https://tools.ietf.org/html/draft-pantos-http-live-streaming-16#section-6.3.3
-    if (!playlist.endList) {
-      liveBuffer = targetDuration * 3;
-      // walk backward from the last available segment and track how
-      // much media time has elapsed until three target durations have
-      // been traversed. if a segment is part of the interval being
-      // reported, subtract the overlapping portion of its duration
-      // from the result.
-      for (i = playlist.segments.length - 1; i >= 0 && liveBuffer > 0; i--) {
-        segment = playlist.segments[i];
-        pending = optionalMin(duration(playlist,
-                                       playlist.mediaSequence + i,
-                                       playlist.mediaSequence + i + 1),
-                           liveBuffer);
-        liveBuffer -= pending;
-        end -= pending;
-      }
-    }
-
+    start = intervalDuration(playlist, playlist.mediaSequence);
+    end = intervalDuration(playlist,
+                           playlist.mediaSequence + playlist.segments.length);
+    end -= (playlist.targetDuration || DEFAULT_TARGET_DURATION) * 3;
```

Comment:
Safari handles live streams differently than the "spec" we should probably not use TARGET_DURATION and just base the calculation on "3 segments from the end" like Safari.